linode/resource_linode_lke_cluster.go

```diff
@@ -604,3 +610,34 @@ func flattenLinodeLKEClusterAPIEndpoints(apiEndpoints []linodego.LKEClusterAPIEn
 	}
 	return flattened
 }
+
+// This cannot currently be handled efficiently by a DiffSuppressFunc
+// See: https://github.com/hashicorp/terraform-plugin-sdk/issues/477
+func matchPoolsWithSchema(pools []linodego.LKEClusterPool, schemaPools []interface{}) []linodego.LKEClusterPool {
+	result := make([]linodego.LKEClusterPool, len(schemaPools))
+
+	poolMap := make(map[int]linodego.LKEClusterPool)
```

Comment:
`poolMap := make(map[int]linodego.LKEClusterPool, len(schemaPools))` to avoid unnecessary reallocation.